build-system/tasks/sweep-experiments/index.js

```diff
@@ -0,0 +1,445 @@
+      prodConfigPath,
+      removeExperiment,
+      percentage
+    );
+    return {[removeExperiment]: {percentage, previousHistory}};
+  }
+
+  const work = {};
+  for (const [experiment, percentage] of Object.entries(prodConfig)) {
+    if (
+      typeof percentage === 'number' &&
+      percentage === canaryConfig[experiment] &&
+      experiment !== 'canary' &&
+      (percentage >= 1 || percentage <= 0)
+    ) {
+      const previousHistory = findConfigBitCommits(
+        cutoffDateFormatted,
+        prodConfigPath,
+        experiment,
+        percentage
+      );
+      if (previousHistory.length > 0) {
+        work[experiment] = {percentage, previousHistory};
+      }
+    }
+  }
+  return work;
+}
+
+async function sweepExperiments() {
+  const headHash = getHeadHash();
+
+  const prodConfig = readJsonSync(prodConfigPath);
+  const canaryConfig = readJsonSync(canaryConfigPath);
+
+  const cutoffDateFormatted = daysAgo(
+    argv.experiment ? 0 : argv.days_ago || 365
+  ).toISOString();
+
+  const work = collectWork(
+    prodConfig,
+    canaryConfig,
+    cutoffDateFormatted,
+    argv.experiment
+  );
+
+  const removed = [];
+
+  let at = 0;
+  const total = Object.keys(work).length;
+
+  if (total === 0) {
+    log(cyan('No experiments to remove.'));
+    log(`Cutoff at ${cutoffDateFormatted}`);
+    return;
+  }
+
+  log();
+  log(cyan('Removing references to the following experiments:'));
+  for (const experiment in work) {
+    log(readableRemovalId(experiment, work[experiment]));
+  }
+  log();
+
+  if (argv.dry) {
+    log('❗️ (Not making changes due to --dry)');
+    return;
+  }
+
+  for (const id in work) {
+    log(`🚮 ${++at}/${total}`, magenta(`${id}...`));
+
+    const modified = [
+      ...removeFromExperimentsConfig(id),
+      ...removeFromJsonConfig(prodConfig, prodConfigPath, id),
+      ...removeFromJsonConfig(canaryConfig, canaryConfigPath, id),
+      ...removeFromRuntimeSource(id, work[id].percentage),
+    ];
+
+    getStdout(
+      `./node_modules/prettier/bin-prettier.js --write ${modified.join(' ')}`
+    );
+
+    gitCommitSingleExperiment(id, work[id], modified)
+      .split('\n')
+      .forEach((line) => log(line));
+
+    log();
+
+    removed.push(`- ${readableRemovalId(id, work[id])}`);
+  }
+
+  if (removed.length > 0) {
+    const modifiedSourceFiles = getModifiedSourceFiles(headHash);
+
+    const htmlFilesWithReferences = filesContainingPattern(
+      containExampleHtml.map((dir) => `${dir}/**/*.html`),
+      `['"](${Object.keys(work).join('|')})['"]`
+    );
+
+    log(
+      getStdout(
+        `git commit --allow-empty -m "${cmdEscape(
+          summaryCommitMessage({
+            removed,
+            modifiedSourceFiles,
+            htmlFilesWithReferences,
+            cutoffDateFormatted: truncateYyyyMmDd(cutoffDateFormatted),
+          })
+        )}"`
+      ),
+      '\n\n',
+      getStdout('git log --pretty=format:%b "HEAD^..HEAD"'),
+      `\n\n`
+    );
```

Comment:
Can this block be replaced by calls to one of the functions in `build-system/common/exec.js`?

Reply:
If you mean `getStdout`, see comment above.

Reply:
No, I meant the functions in `exec.js`, which execute a command with logging instead of  returning its output string. The advantage of `exec*` over `get*` is that all the pretty coloring and formatting of the output is retained.

Reply:
Ah, I see! `exec('git log')` seems to enter an interactive `vim` view, which interrupts the rest of the script logging. It may be better left as-is.

Reply:
Aha, yeah. No worries, and thanks for humoring me!